golem-component-service/db/migration/postgres/001__init.sql

```diff
@@ -1,13 +1,23 @@
 CREATE TABLE components
 (
-    component_id        uuid    NOT NULL,
+    namespace           text    NOT NULL,
+    component_id        uuid    NOT NULL PRIMARY KEY,
     name                text    NOT NULL,
-    size                integer NOT NULL,
+    UNIQUE (namespace, name)
+);
+
+CREATE INDEX components_namespace_id_idx ON components (namespace, component_id);
+
+CREATE TABLE component_versions
+(
+    component_id        uuid    NOT NULL,
     version             bigint  NOT NULL,
+    size                integer NOT NULL,
     created_at          timestamp NOT NULL DEFAULT CURRENT_TIMESTAMP,
     user_component      text    NOT NULL,
     protected_component text    NOT NULL,
     protector_version   bigint,
-    metadata            jsonb   NOT NULL,
-    PRIMARY KEY (component_id, version)
+    metadata            bytea   NOT NULL,
+    PRIMARY KEY (component_id, version),
+    FOREIGN KEY (component_id) REFERENCES components (component_id)
```

Comment:
You can inline it with column declaration this way:
```sql
component_id        uuid    NOT NULL references components (component_id)
```

Reply:
thanks, fixed